PatientWebApplication/UserMicroserviceApi/Service/AbstractUserService.cs

```diff
@@ -0,0 +1,117 @@
+            return true;
+        }
+
+        private bool areUCINsEqual(string firstUCIN, string secondUCIN)
+        {
+            if (firstUCIN.Equals(secondUCIN)) return true;
+            return false;
+        }
+
+        private bool areIDsEqual(int firstID, int secondID)
+        {
+            if (firstID == secondID) return true;
+            return false;
+        }
+
+        private bool areEmailsEqual(string firstEmail, string secondEmail)
+        {
+            if (firstEmail.Equals(secondEmail)) return true;
+            return false;
+        }
+
+        public bool isFoundInManagers(string email, string ucin, T user)
+        {
+            string path = bingPathToAppDir(@"JsonFiles\manager.json");
+            ManagerRepository managerRepository = new ManagerRepository(path);
+            List<ManagerUser> listOfManagers = managerRepository.GetAll();
+
+            foreach (ManagerUser manager in listOfManagers)
+            {
+                if (!areIDsEqual(manager.id, user.id) && (areUCINsEqual(manager.uniqueCitizensidentityNumber, ucin) || areEmailsEqual(manager.email, email)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+
+
+        }
+
+        public bool isFoundInSecretaries(string email, string ucin, T user)
+        {
+            string path = bingPathToAppDir(@"JsonFiles\secretary.json");
+            SecretaryRepository secretaryRepository = new SecretaryRepository(path);
+            List<SecretaryUser> listOfSecretaries = secretaryRepository.GetAll();
+
+            foreach (SecretaryUser secretary in listOfSecretaries)
+            {
+                if (!areIDsEqual(secretary.id, user.id) && (areUCINsEqual(secretary.uniqueCitizensidentityNumber, ucin) || areEmailsEqual(secretary.email, email)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public bool isFoundInDoctors(string email, string ucin, T user)
+        {
+            string path = bingPathToAppDir(@"JsonFiles\doctors.json");
+            DoctorRepository doctorRepository = new DoctorRepository();
+            List<DoctorUser> listOfDoctors = doctorRepository.GetAll();
+
+            foreach (DoctorUser doctor in listOfDoctors)
+            {
+                if (!areIDsEqual(doctor.id, user.id) && (areUCINsEqual(doctor.uniqueCitizensidentityNumber, ucin) || areEmailsEqual(doctor.email, email)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+
+
+        public static string bingPathToAppDir(string localPath)
+        {
+            string currentDir = Environment.CurrentDirectory;
+            DirectoryInfo directory = new DirectoryInfo(
+            Path.GetFullPath(Path.Combine(currentDir, @"..\..\" + localPath)));
+            return directory.ToString();
```

Comment:
Write this in form:
"return (new DirectoryInfo(Path.GetFullPath(Path.Combine(Environment.CurrentDirectory, @"..\..\" + localPath)))).ToString();"